modules/http/http-worker.c

```diff
@@ -512,7 +515,21 @@ _curl_perform_request(HTTPDestinationWorker *self, HTTPLoadBalancerTarget *targe
 
   curl_easy_setopt(self->curl, CURLOPT_URL, target->url);
   curl_easy_setopt(self->curl, CURLOPT_HTTPHEADER, http_curl_header_list_as_slist(self->request_headers));
-  curl_easy_setopt(self->curl, CURLOPT_POSTFIELDS, self->request_body->str);
+  if (owner->message_compression != CURL_COMPRESSION_UNCOMPRESSED)
+    {
+      if(http_dd_compress_string(self->request_body_compressed, self->request_body, owner->message_compression))
+        {
+          curl_easy_setopt(self->curl, CURLOPT_POSTFIELDS, self->request_body_compressed->str);
+          curl_easy_setopt(self->curl, CURLOPT_POSTFIELDSIZE, self->request_body_compressed->len);
+        }
+      else
+        {
+          curl_easy_setopt(self->curl, CURLOPT_POSTFIELDS, self->request_body->str);
+        }
+
+    }
+  else
+    curl_easy_setopt(self->curl, CURLOPT_POSTFIELDS, self->request_body->str);
```

Comment:
If the corresponding `if` has curly braces, please add them to the `else` branch, too.

Reply:
Let's make this note optional as it's a matter of taste.

(Mandatory code style things should be agreed upon and included in our astyle check in my opinion.)

Reply:
Agreed, sorry.